src/compiler/checker.ts

```diff
@@ -3472,25 +3472,41 @@ namespace ts {
             return undefined;
         }
 
-        // Return the inferred type for a variable, parameter, or property declaration
-        function getTypeForJSSpecialPropertyDeclaration(declaration: Declaration): Type {
-            const expression = declaration.kind === SyntaxKind.BinaryExpression ? <BinaryExpression>declaration :
-                declaration.kind === SyntaxKind.PropertyAccessExpression ? <BinaryExpression>getAncestor(declaration, SyntaxKind.BinaryExpression) :
-                undefined;
+        function getWidendedTypeFromJSSpecialPropetyDeclarations(symbol: Symbol) {

```

Comment:
Question: this function is not just for JS right ?( the name got me confused a bit)
Nit: would be great to see JSDoc comment...it is not clear from function name what it try to do